### PR TITLE
NS2.1: System.Security.Cryptography changes

### PR DESCRIPTION
**Pull request by EgorBo**
Contributes to: #11430

Current status:
```
TypeCannotChangeClassification : Type 'System.Decimal' is marked as readonly in the contract so it must also be marked readonly in the implementation.
MembersMustExist : Member 'System.Net.HttpWebResponse..ctor()' does not exist in the implementation but it does exist in the contract.
MembersMustExist : Member 'System.Net.Http.HttpClient.PatchAsync(System.String, System.Net.Http.HttpContent)' does not exist in the implementation but it does exist in the contract.
MembersMustExist : Member 'System.Net.Http.HttpClient.PatchAsync(System.String, System.Net.Http.HttpContent, System.Threading.CancellationToken)' does not exist in the implementation but it does exist in the contract.
MembersMustExist : Member 'System.Net.Http.HttpClient.PatchAsync(System.Uri, System.Net.Http.HttpContent)' does not exist in the implementation but it does exist in the contract.
MembersMustExist : Member 'System.Net.Http.HttpClient.PatchAsync(System.Uri, System.Net.Http.HttpContent, System.Threading.CancellationToken)' does not exist in the implementation but it does exist in the contract.
MembersMustExist : Member 'System.Net.Http.HttpClientHandler.DangerousAcceptAnyServerCertificateValidator.get()' does not exist in the implementation but it does exist in the contract.
MembersMustExist : Member 'System.Net.Http.HttpMethod.Patch.get()' does not exist in the implementation but it does exist in the contract.
TypesMustExist : Type 'System.Net.Http.ReadOnlyMemoryContent' does not exist in the implementation but it does exist in the contract.
Total Issues: 9
```

### CODE REVIEW

**Comment by EgorBo** [comment]
@monojenkins build Windows x64 C++

File: mcs/class/System.Core/System.Security.Cryptography.X509Certificates/RSACertificateExtensions.cs [hunk 1]
```diff
@@ -47,5 +47,10 @@ public static RSA GetRSAPublicKey(this X509Certificate2 certificate)
 				throw new ArgumentNullException("certificate");
 			return certificate.PublicKey.Key as RSA;
 		}
+		
+		public static X509Certificate2 CopyWithPrivateKey(this X509Certificate2 certificate, RSA privateKey) 
+		{
```
Comment (baulig) [inline mcs/class/System.Core/System.Security.Cryptography.X509Certificates/RSACertificateExtensions.cs]: This is not too difficult, I'll take care of that.

Comment (baulig) [reply mcs/class/System.Core/System.Security.Cryptography.X509Certificates/RSACertificateExtensions.cs]: @EgorBo I'll take care of this in https://github.com/mono/mono/pull/13037.

**Comment by EgorBo** [comment]
@monojenkins build Linux AArch64 Coop Suspend

**Comment by marek-safar** [comment]
@EgorBo could you please resolve the conflict

**Comment by marek-safar** [comment]
@monojenkins commit csproj and apidiff

File: mcs/class/System.Core/System.Security.Cryptography.X509Certificates/RSACertificateExtensions.cs (hunk 1 above)
Comment (baulig) [reply mcs/class/System.Core/System.Security.Cryptography.X509Certificates/RSACertificateExtensions.cs]: Unfortunately, this is a lot more difficult than I expected, so I think we should keep the exception there for the moment.

**Pull request merged**